src/components/modal/index.stories.mdx

```diff
@@ -0,0 +1,150 @@
+import { Canvas, Meta, Story } from '@storybook/addon-docs';
+import { action } from '@storybook/addon-actions';
+import { useDisclosure } from '@chakra-ui/react';
+
+import Button from '../button';
+import Box from '../box';
+
+import Modal from './index';
+
+<Meta title="Components/Data display/Modal" component={Modal} />
+
+export const ModalTemplate = (args) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  return (
+    <Box>
+      <Button onClick={onOpen}>Open</Button>
+      <Modal isOpen={isOpen} onClose={onClose} {...args}>
+        <Box p="4">
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer
+          elementum ultricies vestibulum. Sed odio diam, tristique id velit sit
+          amet, sodales aliquet magna. Fusce vitae orci vel nulla vehicula
+          condimentum. Maecenas sollicitudin volutpat tellus, laoreet fringilla
+          nibh. Vivamus luctus enim id quam feugiat, maximus porttitor justo
+          ultrices. Nam tellus erat, porta nec consectetur eget, aliquet id
+          purus. Suspendisse at elementum arcu. Suspendisse ac lobortis velit.
+          Phasellus malesuada egestas est sit amet venenatis.
+        </Box>
+      </Modal>
+    </Box>
+  );
+};
+
+# Modal
+
+<Canvas>
+  <Story
+    name="Modal"
+    args={{
+      isCentered: true,
+      size: 'md',
+      motionPreset: 'scale',
+      title: 'Modal Header',
+    }}
+    argTypes={{
+      size: {
+        options: ['md', 'full'],
+        control: 'select',
+      },
+      motionPreset: {
+        options: ['slideInBottom', 'slideInRight', 'scale'],
+        control: 'select',
+      },
```

Comment:
could you disable the action prop for the buttons in the argTypes? because it looks like this: 
![image](https://user-images.githubusercontent.com/71456764/208689730-cabfe710-e9f1-4fb1-a0fc-63b8dca1cafe.png)
 you can check in show/hide for example how you can disable certain props in storybook